--- ../src_base/minecraft/net/minecraft/src/BiomeGenBase.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src_work/minecraft/net/minecraft/src/BiomeGenBase.java	0000-00-00 00:00:00.000000000 -0000
@@ -102,7 +102,7 @@
     protected WorldGenForest worldGenForest;
     protected WorldGenSwamp worldGenSwamp;
 
-    protected BiomeGenBase(int par1)
+    public BiomeGenBase(int par1)
     {
         this.topBlock = (byte)Block.grass.blockID;
         this.fillerBlock = (byte)Block.dirt.blockID;
@@ -139,7 +139,7 @@
     /**
      * Allocate a new BiomeDecorator for this BiomeGenBase
      */
-    protected BiomeDecorator createBiomeDecorator()
+    public BiomeDecorator createBiomeDecorator()
     {
         return new BiomeDecorator(this);
     }
@@ -147,7 +147,7 @@
     /**
      * Sets the temperature and rainfall of this biome.
      */
-    private BiomeGenBase setTemperatureRainfall(float par1, float par2)
+    public BiomeGenBase setTemperatureRainfall(float par1, float par2)
     {
         if (par1 > 0.1F && par1 < 0.2F)
         {
@@ -164,7 +164,7 @@
     /**
      * Sets the minimum and maximum height of this biome. Seems to go from -2.0 to 2.0.
      */
-    private BiomeGenBase setMinMaxHeight(float par1, float par2)
+    public BiomeGenBase setMinMaxHeight(float par1, float par2)
     {
         this.minHeight = par1;
         this.maxHeight = par2;
@@ -174,7 +174,7 @@
     /**
      * Disable the rain for the biome.
      */
-    private BiomeGenBase setDisableRain()
+    public BiomeGenBase setDisableRain()
     {
         this.enableRain = false;
         return this;
@@ -196,25 +196,25 @@
     /**
      * sets enableSnow to true during biome initialization. returns BiomeGenBase.
      */
-    protected BiomeGenBase setEnableSnow()
+    public BiomeGenBase setEnableSnow()
     {
         this.enableSnow = true;
         return this;
     }
 
-    protected BiomeGenBase setBiomeName(String par1Str)
+    public BiomeGenBase setBiomeName(String par1Str)
     {
         this.biomeName = par1Str;
         return this;
     }
 
-    protected BiomeGenBase func_4124_a(int par1)
+    public BiomeGenBase func_4124_a(int par1)
     {
         this.field_6502_q = par1;
         return this;
     }
 
-    protected BiomeGenBase setColor(int par1)
+    public BiomeGenBase setColor(int par1)
     {
         this.color = par1;
         return this;
