--- ../src_base/minecraft/net/minecraft/src/BlockCactus.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src_work/minecraft/net/minecraft/src/BlockCactus.java	0000-00-00 00:00:00.000000000 -0000
@@ -2,6 +2,8 @@
 
 import java.util.Random;
 
+import net.minecraft.src.BAPI.Main;
+
 public class BlockCactus extends Block
 {
     protected BlockCactus(int par1, int par2)
@@ -135,6 +137,10 @@
         {
             return false;
         }
+        else if(Main.canPlaceC(par1World.getBlockId(par2, par3 - 1, par4), par1World.getBlockMetadata(par2, par3 - 1, par4)))
+        {
+        	return true;
+        }
         else
         {
             int var5 = par1World.getBlockId(par2, par3 - 1, par4);
