--- ../src_base/minecraft/net/minecraft/src/GenLayerBiome.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src_work/minecraft/net/minecraft/src/GenLayerBiome.java	0000-00-00 00:00:00.000000000 -0000
@@ -1,8 +1,10 @@
 package net.minecraft.src;
 
+import net.minecraft.src.BAPI.Main;
+
 public class GenLayerBiome extends GenLayer
 {
-    public static BiomeGenBase[] biomeArray = new BiomeGenBase[] {BiomeGenBase.desert, BiomeGenBase.forest, BiomeGenBase.extremeHills, BiomeGenBase.swampland, BiomeGenBase.plains, BiomeGenBase.taiga, BiomeGenBase.jungle};
+    public static BiomeGenBase[] biomeArray = (BiomeGenBase[]) Main.getBiome(1).toArray(new BiomeGenBase[Main.getBiome(1).size()]);
 
     /** this sets all the biomes that are allowed to appear in the overworld */
     private BiomeGenBase[] allowedBiomes;
@@ -15,7 +17,7 @@
 
         if (par4WorldType == WorldType.DEFAULT_1_1)
         {
-            this.allowedBiomes = new BiomeGenBase[] {BiomeGenBase.desert, BiomeGenBase.forest, BiomeGenBase.extremeHills, BiomeGenBase.swampland, BiomeGenBase.plains, BiomeGenBase.taiga};
+            this.allowedBiomes = (BiomeGenBase[]) Main.getBiome(5).toArray(new BiomeGenBase[Main.getBiome(5).size()]);
         }
     }
 
