--- ../src_base/minecraft/net/minecraft/src/GuiWinGame.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src_work/minecraft/net/minecraft/src/GuiWinGame.java	0000-00-00 00:00:00.000000000 -0000
@@ -6,6 +6,9 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Random;
+
+import net.minecraft.src.BAPI.Main;
+
 import org.lwjgl.opengl.GL11;
 
 public class GuiWinGame extends GuiScreen
@@ -109,6 +112,8 @@
 
                 var4 = new BufferedReader(new InputStreamReader(GuiWinGame.class.getResourceAsStream("/title/credits.txt"), Charset.forName("UTF-8")));
 
+                Main.handleCredits(lines);
+                
                 while ((var1 = var4.readLine()) != null)
                 {
                     var1 = var1.replaceAll("PLAYERNAME", this.mc.session.username);
