--- ../src_base/minecraft/net/minecraft/src/MapGenVillage.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src_work/minecraft/net/minecraft/src/MapGenVillage.java	0000-00-00 00:00:00.000000000 -0000
@@ -4,10 +4,12 @@
 import java.util.List;
 import java.util.Random;
 
+import net.minecraft.src.BAPI.Main;
+
 public class MapGenVillage extends MapGenStructure
 {
     /** A list of all the biomes villages can spawn in. */
-    public static List villageSpawnBiomes = Arrays.asList(new BiomeGenBase[] {BiomeGenBase.plains, BiomeGenBase.desert});
+    public static List villageSpawnBiomes = Main.getBiome(1);
 
     /** World terrain type, 0 for normal, 1 for flat map */
     private final int terrainType;
