--- ../src_base/minecraft/net/minecraft/src/SaveHandler.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src_work/minecraft/net/minecraft/src/SaveHandler.java	0000-00-00 00:00:00.000000000 -0000
@@ -9,6 +9,8 @@
 import java.util.List;
 import java.util.logging.Logger;
 
+import net.minecraft.src.BAPI.Main;
+
 public class SaveHandler implements ISaveHandler
 {
     /** Reference to the logger. */
@@ -121,6 +123,10 @@
      */
     public WorldInfo loadWorldInfo()
     {
+        File file;
+        file = new File(saveDirectory, "BAPI");
+        file.mkdirs();
+        Main.loadNBT(file);
         File var1 = new File(this.saveDirectory, "level.dat");
         NBTTagCompound var2;
         NBTTagCompound var3;
@@ -163,6 +169,10 @@
      */
     public void saveWorldInfoAndPlayer(WorldInfo par1WorldInfo, List par2List)
     {
+        File file;
+        file = new File(saveDirectory, "BAPI");
+        file.mkdirs();
+        Main.saveNBT(file);
         NBTTagCompound var3 = par1WorldInfo.getNBTTagCompoundWithPlayers(par2List);
         NBTTagCompound var4 = new NBTTagCompound();
         var4.setTag("Data", var3);
